src/kani-compiler/cbmc/src/goto_program/expr.rs

```diff
@@ -851,13 +852,10 @@ impl Expr {
             // Floating Point Equalities
             IeeeFloatEqual | IeeeFloatNotequal => lhs.typ == rhs.typ && lhs.typ.is_floating_point(),
             // Overflow flags
-            OverflowMinus => {
-                (lhs.typ == rhs.typ && (lhs.typ.is_pointer() || lhs.typ.is_numeric()))
-                    || (lhs.typ.is_pointer() && rhs.typ.is_integer())
-            }
-            OverflowMult | OverflowPlus => {
-                (lhs.typ == rhs.typ && lhs.typ.is_integer())
-                    || (lhs.typ.is_pointer() && rhs.typ.is_integer())
+            // While one can technically use these on pointers, the result is treated as an integer.
+            // This is almost never what you actually want.
+            OverflowMinus | OverflowMult | OverflowPlus => {
+                lhs.typ.is_integer() && rhs.typ.is_integer()
```

Comment:
Suggestion: Follow the same style as in other arms.
```suggestion
                (lhs.typ == rhs.typ && lhs.typ.is_integer())
```

Reply:
done